src/components/Auth/ResetPassword.tsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { useState, useCallback } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Form, Typography, Input, Button, notification } from 'antd';
+import styled from 'styled-components';
+import qs from 'query-string';
+
+import { useAuth } from '../../hooks';
+import { UserAPI } from '../../api';
+import { validatePassword } from './Auth';
+
+function ResetPassword() {
+  const [form] = Form.useForm();
+  const { authenticating } = useAuth();
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+
+  const history = useHistory();
+
+  const q = qs.parse(document.location.search);
+  const token = q.token as string;
+  const login = (q.email as string) || '';
+
+  const handlePasswordChange = useCallback((e: any) => {
+    setPassword(e.target.value);
+  }, []);
+  const handleconfirmPasswordChange = useCallback((e: any) => {
+    setConfirmPassword(e.target.value);
+  }, []);
+
+  const showResetSuccessNotification = () => {
+    notification.success({
+      message: 'Password has been successfully changed.',
+      description: 'Your password has been successfully changed. You can log in with your new password now.',
+      duration: 3.5,
+    });
+  };
+
+  const handlePasswordReset = useCallback(async () => {
+    await form.validateFields();
+    history.push('/login/');
+    const resetPassword = await UserAPI.resetPassword({ login, password, token });
+    if (resetPassword) {
+      showResetSuccessNotification();
+    }
+  }, [login, token, password, form, history]);
+
+  async function checkPasswordMatch() {
+    if (!(password === confirmPassword)) {
+      return Promise.reject(new Error('Passwords must match'));
+    }
+  }
+
+  return (
+    <Container>
+      <Form form={form}>
+        <Typography.Title level={3} className="title">
+          {`Reset Password ${login.replace(/\s/g, '')}`}
+        </Typography.Title>
+        <Form.Item
+          name="password"
+          rules={[
+            { required: true, message: 'Password is required' },
+            { validator: (_, value) => validatePassword(value) },
+          ]}
+        >
+          <Input
+            type="password"
+            size="large"
+            style={{ width: '240px' }}
```

Comment:
duplicated styling + you should avoid using `style`. just use a common `className` for both inputs like `passwordInput` and put the style there